.gitignore

```diff
@@ -0,0 +1,32 @@
+# output files
```

Comment:
Is this a new file? 
@rfalgout @cadaverous-lives 
Do we want to distribute a default gitignore file?  I'm a little on the fence here.  Could this interfere with someone's existing .gitignore?

Reply:
I didn't really intend to leave my gitignore in there, sorry! I was just using it to simplify my own usage.

Reply:
We do include a `.gitignore` file in hypre now: https://github.com/hypre-space/hypre/blob/master/.gitignore.  It's a good way to prevent certain files from accidentally being committed, and convenient for developers.  It does not override a global `.gitignore`, it adds to it.  If we do include it here, we should at least discuss its contents.  That's my 2 cents.

Reply:
I could see including a .gitignore being beneficial.  Often, my git status is swamped with temporary files.  

David: I think if you can add a couple comments explaining the contents, including the file isn't a bad thing.